ci/test_wheel.sh

```diff
@@ -6,8 +6,20 @@ set -eou pipefail
 RAPIDS_PY_CUDA_SUFFIX="$(rapids-wheel-ctk-name-gen ${RAPIDS_CUDA_VERSION})"
 WHEELHOUSE="${PWD}/dist/"
 RAPIDS_PY_WHEEL_NAME="rmm_${RAPIDS_PY_CUDA_SUFFIX}" rapids-download-wheels-from-s3 python "${WHEELHOUSE}"
+PIP_PACKAGE=$(echo "${WHEELHOUSE}"/rmm_"${RAPIDS_PY_CUDA_SUFFIX}"*.whl | head -n1)
 
-# echo to expand wildcard before adding '[extra]' requires for pip
-python -m pip install -v "$(echo "${WHEELHOUSE}"/rmm_${RAPIDS_PY_CUDA_SUFFIX}*.whl)[test]"
+# Use `package[test]` to install latest test dependencies or explicitly install oldest.
+if [[ $RAPIDS_DEPENDENCIES != "oldest" ]]; then
+  python -m pip install -v "${PIP_PACKAGE}[test]"
+else
+  python -m pip install rapids-dependency-file-generator && pyenv rehash  # TODO: include in image
+  rapids-dependency-file-generator \
+      --output requirements \
+      --file-key test_python \
+      --matrix "cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch);py=${RAPIDS_PY_VERSION};dependencies=${RAPIDS_DEPENDENCIES}" \
+    | tee oldest-dependencies.txt
```

Comment:
Ok, this works now also for the wheel tests.  A bit unfortunate that we need the branch and maybe the `dependencies.yaml` entries.

On the up-side, it should be impossible to forgot to update the dependency file, as the below install will fail if there is an inconsistency:
```
ERROR: Cannot install numpy==1.23 and numpy>1.25 because these package versions have conflicting dependencies.

The conflict is caused by:
    The user requested numpy>1.25
    The user requested numpy==1.23
```

Reply:
I think this could be simplified a bit, and that it'd be better to use **constraints** instead of **requirements**:

* so we don't end up installing anything unnecessary
* so this test can still catch issues of the form "wheel forgot to include some required dependencies in its metadata

Would you consider something like this instead?

```shell
echo "" > ./constraints.txt
if [[ $RAPIDS_DEPENDENCIES == "oldest" ]]; then
  rapids-dependency-file-generator \
        --output requirements \
        --file-key test_python \
        --matrix "cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch);py=${RAPIDS_PY_VERSION};dependencies=${RAPIDS_DEPENDENCIES}" \
      | tee ./constraints.txt
fi

# echo to expand wildcard before adding '[extra]' requires for pip
python -m pip install \
    -v \
    --constraint ./constraints.txt \
    "$(echo "${WHEELHOUSE}"/rmm_${RAPIDS_PY_CUDA_SUFFIX}*.whl)[test]"
```